src/prompt.h

```diff
@@ -40,23 +40,21 @@ typedef struct FriendRequests {
     struct friend_request request[MAX_FRIEND_REQUESTS];
 } FriendRequests;
 
-extern ToxWindow *prompt;
 extern FriendRequests FrndRequests;
 
 ToxWindow *new_prompt(void);
 
 void prep_prompt_win(void);
-void prompt_init_statusbar(ToxWindow *self, Toxic *toxic, bool first_time_run);
-void prompt_update_nick(ToxWindow *prompt, const char *nick);
-void prompt_update_statusmessage(ToxWindow *prompt, Toxic *toxic, const char *statusmsg);
-void prompt_update_status(ToxWindow *prompt, Tox_User_Status status);
-void prompt_update_connectionstatus(ToxWindow *prompt, bool is_connected);
-void kill_prompt_window(ToxWindow *self, const Client_Config *c_config);
+void prompt_init_statusbar(Toxic *toxic, bool first_time_run);
+void prompt_update_nick(Toxic *toxic, const char *nick);
```

Comment:
Not sure when, but for some things it might make sense to pass what parts of the Toxic object they need rather than giving them everything?

Always passing all state mutably to every function might make reasoning about where things are used and when they might be changed difficult. E.g. passing a `Client_Config const *` means that a function only reads settings, but a `Client_Config *` says that a function loads or sets settings.

Reply:
I intentionally changed that to take a Toxic object instead of a ToxWindow. I was debating it in my head for a while and decided that since all prompt_ functions should only ever print to the prompt window, we shouldn't have the option to pass them a non-prompt ToxWindow. I agree with your reasoning in general though. I'm still not sure about this one.

Reply:
Upon further reflection, I think you're right, it was better the way it was. Reverted.